### DISCUSSION TITLE
Understanding Token Balance Cost for GPT-4

### DISCUSSION THREAD

**Discussion by SohelKabir** [danny-avila/LibreChat#1777] [Q&A] [answered]
### What happened?

I have given a prompt to GPT_4: "write a article about ai in 1000 words",
OpenAI response text was around 827 Tokens and 4688 Characters (Checked on https://platform.openai.com/tokenizer).

But system deducted around 50000 (fifty thousands) tokens from my balance!

### Steps to Reproduce

Use GPT_-4 and give prompt ""write a article about ai in 1000 words""

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by SohelKabir** [comment]
Funny thing is only this prompt causes this, when i use write an article about something it doesn't caused the issue even the though response text is almost same sizes in words/tokens/characters !!

**Comment by danny-avila** [comment, marked as answer]
The balance is not synonymous with tokens. It is associated with cost. GPT-4 completion tokens get charged at a rate of 60 (that is, $0.06 / 1K tokens) so that would be 49,620 credits, or $0.04962, so what you said about 50000 (fifty thousands) ~~tokens~~ credits is accurate for 827 completion tokens generated.

Here is a more detailed write up of what your balance represents: https://github.com/danny-avila/LibreChat/discussions/1640#discussioncomment-8251970

**Reply by SohelKabir** [reply]
Thank you for responding.
Can you give me simple idea for this use case?
If a user recharge 1 USD in our system how much credit/balance should we give him?

**Reply by SohelKabir** [reply]
My current understanding is for  1 USD  we should give user  =  (1 USD / 0.001) * 1000  =  1000000 Credit

**Reply by danny-avila** [reply]
Yup! Each million credit is $1 USD
